dashboard/app/models/user.rb

```diff
@@ -2877,6 +2877,8 @@ def self.us_state_dropdown_options
       errors.add(:us_state, :blank)
       return
     end
+    # Check if us_state value will change, check after making sure us_state is not blank
+    return unless will_save_change_to_properties? && properties_change&.first&.[]("us_state") != us_state
```

Comment:
I'm guessing `us_state_changed?` doesn't work because it is a serialized attribute?
The `.first` is concerning to me because I'm wondering what is `.second` or `.last`. I'm not familiar with `properties_change`.

Reply:
Correct, it was a bit tricky because of `us_state` being part of serialized attributes.
I used the documentation provided[ here](https://api.rubyonrails.org/classes/ActiveModel/Dirty.html#method-i-2A_change) for` _change` which Returns the old and the new value of the attribute.